ale_linters/go/gobuild.vim

```diff
@@ -116,42 +104,116 @@ function! ale_linters#go#gobuild#GetCommand(buffer, copy_output) abort
       continue
     endif
 
-    " only consider buffers other than a:buffer if they have the same import
-    " path as a:buffer and are modified
-    if l:bufnum != a:buffer
-      if s:PackageImportPath(l:bufnum) !=# l:importpath
-        continue
-      endif
+    " only consider buffers if they have the same import path as a:buffer and
+    " are modified
 
-      if !getbufvar(l:bufnum, '&mod')
-        continue
-      endif
+    let l:bufpkg = ale_linters#go#gobuild#PackageImportPath(l:bufnum)
+    if index(l:importpaths, l:bufpkg) == -1
+      continue
+    endif
+
+    if !getbufvar(l:bufnum, '&mod')
+      continue
     endif
 
-    call writefile(getbufline(l:bufnum, 1, '$'), l:tempdir . '/src/' . s:PkgFile(l:bufnum))
+    call add(l:output, l:bufnum)
   endfor
 
-  let l:gopaths = [ l:tempdir ]
-  call extend(l:gopaths, split(s:go_env.GOPATH, s:SplitChar))
+  return l:output
+endfunction
 
-  return 'GOPATH=' . shellescape(join(l:gopaths, s:SplitChar)) . ' go test -c -o /dev/null ' . shellescape(l:importpath)
+function! s:SymlinkFilesCmd(srcdir, destdir, files) abort
+  " symlink the files to a temp directory with $GOPATH structure
+  let l:cmds = []
+
+  for l:file in a:files
+    " TODO(jrubin) this will not work on windows
+    " TODO(jrubin) no idea if this works on windows or if this will work on
+    " shells like csh, ksh, fish, etc.
+    call add(l:cmds, 'ln -s ' . shellescape(a:srcdir . '/src/' . l:file) . ' ' . shellescape(a:destdir . '/src/' . l:file))
```

Comment:
`cp` wouldn't have worked on Windows anyway, I just realised. I wonder if a copy should be made on Windows, and a symlink should be used on Unix systems.